src/completion.ts

```diff
@@ -104,18 +105,24 @@ export function asCompletionItem(entry: tsp.CompletionEntry, file: string, posit
             }
         }
     }
+    if (!insertText) {
+        insertText = item.label;
+    }
     if (replacementRange) {
-        if (!insertText) {
-            insertText = item.label;
-        }
-
-        item.textEdit = lsp.TextEdit.replace(replacementRange, insertText);
-    } else {
-        item.insertText = insertText;
+        item.textEdit = createTextEdit(replacementRange, position, insertText, features);
     }
+    item.insertText = insertText;
     return item;
 }
 
+function createTextEdit(replacementRange: lsp.Range, position: lsp.Position, insertText: string, features: SupportedFeatures): lsp.TextEdit | lsp.InsertReplaceEdit {
+    if (features.insertReplaceSupport) {
+        const insertRange: lsp.Range = { start: replacementRange.start, end: position };
```

Comment:
I'm not sure there is really a need for a separate function for this but it would be simpler to use helper function here.

```suggestion
        const insertRange = lsp.Range.create(replacementRange.start, position);
```